app/src/test/kotlin/com/intive/picover/profile/delete/viewmodel/DeleteAccountViewModelTest.kt

```diff
@@ -0,0 +1,37 @@
+package com.intive.picover.profile.delete.viewmodel
+
+import com.intive.picover.auth.model.AccountDeletionResult
+import com.intive.picover.auth.repository.AuthRepository
+import com.intive.picover.common.coroutines.CoroutineTestExtension
+import com.intive.picover.profile.delete.model.DeleteAccountEvent
+import com.intive.picover.profile.delete.model.DeleteAccountSideEffect
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.inspectors.forAll
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+
+class DeleteAccountViewModelTest : ShouldSpec(
+	{
+		extension(CoroutineTestExtension())
+
+		val authRepository: AuthRepository = mockk(relaxed = true)
+		val tested by lazy { DeleteAccountViewModel(authRepository) }
+
+		should("call deleteAccount on AuthRepository AND set effect depending on result WHEN DeleteAccount event") {
+			listOf(
+				AccountDeletionResult.Success to DeleteAccountSideEffect.ShowSuccessMessage,
+				AccountDeletionResult.ReAuthenticationNeeded to DeleteAccountSideEffect.ShowReAuthenticationNeededMessage,
+			).forAll { (result, sideEffect) ->
+				coEvery { authRepository.deleteAccount() } returns result
+
+				tested.handleEvent(DeleteAccountEvent.DeleteAccount)
```

Comment:
Check: 
1. https://github.com/Android-Guild/Picover/pull/413/files#r1391549154
1. https://github.com/Android-Guild/Picover/pull/413#discussion_r1391591834

Reply:
I will correct my PR after merging #413

Reply:
@BorysChajdas #413 is already merged :) No pressure thou just FYI :)